### PR TITLE
feat: add button link component

### LINKED ISSUE #107: [Button] Can't specify 'as' property directly on Button component
We should be able to specify an `as` property on a component in a way that it's rendered using a different Primitive that it usually should use. For some reason, it's not possible to do this in some places.

### PR DESCRIPTION
**Pull request by seedy**
### Description

We have use cases for `Button`s which hold `href`

- Added a `LinkButton` component
- Added a story inside Button folder

Do you think it's a good idea adding this component to Faency?

### Accessibility concerns

Links and Buttons should not be mixed together

### RFC

RFC proposal issue: https://github.com/traefik/faency/issues/218

### Close issues

If agreed upon, this PR would close #107

### CODE REVIEW

File: components/Button/Button.tsx [hunk 1]
```diff
@@ -234,8 +234,12 @@ export const StyledButton = styled('button', {
     size: 'medium',
     variant: 'primary',
   },
-});
+}
+
+export const StyledLinkButton = styled('a', buttonStyles);
```
[the remark this answers is not in the document]
Comment (seedy) [reply components/Button/Button.tsx:239]: @paulocfjunior I just noticed you have `state="active"` on the ButtonLink record.

`state="active"` impacts button behaviour.

I'm updating the `state` control to reset `state` to undefined

**Review by matthieuh — APPROVED** [review]
LGTM

**Review by matthieuh — COMMENTED** [review]
_deleted comment: somehow GH is approving PRs only adding a review comment 🤔_

**Review by matthieuh — COMMENTED** [review]
How is it different from using <Button as="a" ... syntax ?
In the future, I would suggest having proposals as RFC issues to avoid starting any implementation before we validate them as a team.

**Review by matthieuh — CHANGES_REQUESTED** [review]
How is it different from using <Button as="a" ... syntax?
In the future, I would suggest having proposals as RFC issues to avoid starting any implementation before we validate them as a team.

**Comment by seedy** [comment]
> How is it different from using <Button as="a" ... syntax? In the future, I would suggest having proposals as RFC issues to avoid starting any implementation before we validate them as a team.

@matthieuh 

I agree with you. As you can read in this [RFC proposal](https://github.com/traefik/faency/issues/218), as the pattern is deprecated by Radix-UI, I knew it was very unlikely that we wouldn't follow it.

Next time, I'll wait for RFC review.

**Review by paulocfjunior — APPROVED** [review]

**Pull request closed without merging**